src/System.Threading.AccessControl/src/System/Threading/EventWaitHandleAcl.cs

```diff
@@ -0,0 +1,117 @@
+            }
+            if (eventSecurity == null)
+            {
+                throw new ArgumentNullException(nameof(eventSecurity));
+            }
+
+            bool isManualReset = mode switch
+            {
+                EventResetMode.ManualReset => true,
+                EventResetMode.AutoReset => false,
+                _ => throw new ArgumentOutOfRangeException(nameof(mode), SR.Format(SR.ArgumentOutOfRange_Enum))
+            };
+
+            return CreateEventWaitHandle(initialState, isManualReset, name, out createdNew, eventSecurity);
+        }
+
+        private static unsafe EventWaitHandle CreateEventWaitHandle(bool initialState, bool isManualReset, string name, out bool createdNew, EventWaitHandleSecurity security)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(name));
+            Debug.Assert(security != null);
+
+            EventWaitHandle eventHandle = null;
+
+            fixed (byte* pSecurityDescriptor = security.GetSecurityDescriptorBinaryForm())
+            {
+                var secAttrs = new Interop.Kernel32.SECURITY_ATTRIBUTES
+                {
+                    nLength = (uint)sizeof(Interop.Kernel32.SECURITY_ATTRIBUTES),
+                    lpSecurityDescriptor = (IntPtr)pSecurityDescriptor
+                };
+
+                SafeWaitHandle handle = Interop.Kernel32.CreateEvent(ref secAttrs, isManualReset, initialState, name);
+                ValidateHandle(handle, name, out createdNew);
+
+                AggregateException aggEx = null;
+
+                // Ensure we always close the handle regardless of the OpenExisting call result
+                try
+                {
+                    eventHandle = EventWaitHandle.OpenExisting(name);
+                }
+                catch (Exception ex)
+                {
+                    aggEx = new AggregateException(ex);
+                }
+                finally
+                {
+                    handle.Dispose();
+                }
+
+                if (aggEx != null)
+                {
+                    throw aggEx;
+                }
+            }
+
+            return eventHandle;
+        }
+
+        private static void ValidateHandle(SafeWaitHandle handle, string name, out bool createdNew)
+        {
+            Debug.Assert(handle != null);
+
+            int errorCode = Marshal.GetLastWin32Error();
+
+            if (handle.IsInvalid)
+            {
+                handle.SetHandleAsInvalid();
+
+                if (!string.IsNullOrEmpty(name) && errorCode == Interop.Errors.ERROR_INVALID_HANDLE)
+                    throw new WaitHandleCannotBeOpenedException(SR.Format(SR.WaitHandleCannotBeOpenedException_InvalidHandle, name));
+
+                throw Win32Marshal.GetExceptionForWin32Error(errorCode, name);
+            }
+
+            createdNew = errorCode != Interop.Errors.ERROR_ALREADY_EXISTS;
```

Comment:
nit, `createdNew = (errorCode != Interop.Errors.ERROR_ALREADY_EXISTS);`

Reply:
It's nice to avoid `out` parameters if possible as they don't compose well (and possibly don't generate such good code)_. Although, it might be unclear if it was the return value.